src/Response/PsrResponse.php

```diff
@@ -0,0 +1,30 @@
+<?php
+namespace ByJG\ApiTools\Response;
+
+use Psr\Http\Message\ResponseInterface as PsrResponseInterface;
+
+class PsrResponse implements ResponseInterface
+{
+    protected $interface;
+
+    public function __construct(PsrResponseInterface $interface)
+    {
+        $this->interface = $interface;
+    }
+
+    public function getHeaders()
+    {
+        return $this->interface->getHeaders();
+    }
+
+    public function getStatusCode()
+    {
+        return $this->interface->getStatusCode();
+    }
+
+    public function getBody()
+    {
+        return $this->interface->getBody();
+    }
+
+}
```

Comment:
This is the only class implementing `ResponseInterface`, so I wonder why there is a separate interface at all. Also, it is just a wrapper around the existing interface, so it doesn't add any feature but just a layer of code.

Concerning this part of your proposed changes, I'd rather not merge them. BTW: I'm using this framework with Phalcon, which also doesn't have a PSR request and response type. In order to achieve that, we create the Phalcon type from the PSR type (simple conversion) and then convert the response in a similar way. We also use this in a way that no actual HTTP transfer is made, the request is directly injected into a kernel in-process. Is there a reason this doesn't work for you?

Reply:
I understand, I was able to do a simple "swap" of the requester, that's why I wanted separate layer to add ability to use "anything". Maybe it would be better not to implement `responseInterface`.

Here I'm converting Laravel response to interface that AbstractRequester needs:

https://github.com/pionl/laravel-swagger-test/blob/master/src/Response/LaravelResponse.php

Now I'm able to change handleRequest implementation and swap it with Laravel request:

https://github.com/pionl/laravel-swagger-test/blob/master/src/LaravelRequester.php

Reply:
Of course I can build psr interface but `BadResponseException` will not be thrown (I would have to throw it -> more dependency on guzzle.